File: backend/app/schemas/ai.py
from __future__ import annotations
import uuid
from pydantic import BaseModel, Field


class MessageIn(BaseModel):
    role: str = Field(..., pattern="^(user|assistant)$")
    content: str


class ChatRequest(BaseModel):
    message: str = Field(..., min_length=1, max_length=4000)
    conversation_id: str | None = None
    experiment_context: str | None = None  # optional summary passed from frontend


class ConversationResponse(BaseModel):
    id: str
    title: str | None
    message_count: int
    created_at: str
    updated_at: str


class MessageResponse(BaseModel):
    id: str
    role: str
    content: str
    created_at: str


class ConversationDetailResponse(BaseModel):
    id: str
    title: str | None
    messages: list[MessageResponse]

File: frontend/src/types/ai.ts
export interface ChatMessage {
  id: string;
  role: "user" | "assistant";
  content: string;
  createdAt?: string;
  isStreaming?: boolean;
}

export interface Conversation {
  id: string;
  title: string | null;
  message_count: number;
  created_at: string;
  updated_at: string;
}

export interface ConversationDetail {
  id: string;
  title: string | null;
  messages: ChatMessage[];
}

export interface ChatRequest {
  message: string;
  conversation_id?: string;
  experiment_context?: string;
}

export type StreamEvent =
  | { type: "meta"; conversation_id: string }
  | { type: "chunk"; text: string }
  | { type: "done" }
  | { type: "error"; message: string };
